cron.php

```diff
@@ -0,0 +1,464 @@
+	 * Display an error if the plugin's conditions aren't met
+	 */
+	public function admin_notice() {
+		?>
+		<div class="notice notice-error">
+			<p><?php printf( __( '<strong>%1$s</strong>: To use this plugin, define the constant %2$s.', 'wp-cron-control-revisited' ), 'WP-Cron Control Revisited', '<code>WP_CRON_CONTROL_SECRET</code>' ); ?></p>
+		</div>
+		<?php
+	}
+
+	/**
+	 * PLUGIN FUNCTIONALITY
+	 */
+
+	/**
+	 * List events pending for the current period
+	 */
+	public function get_events() {
+		$events = get_option( 'cron' );
+
+		// That was easy
+		if ( ! is_array( $events ) || empty( $events ) ) {
+			return rest_ensure_response( array( 'events' => null, ) );
+		}
+
+		// To be safe, re-sort the array just as Core does when events are scheduled
+		// Ensures events are sorted chronologically
+		uksort( $events, 'strnatcasecmp' );
+
+		// Select only those events to run in the next sixty seconds
+		// Will include missed events as well
+		$current_events = $internal_events = array();
+		$current_window = strtotime( sprintf( '+%d seconds', $this->job_queue_window_in_seconds ) );
+
+		foreach ( $events as $timestamp => $timestamp_events ) {
+			// Skip non-event data that Core includes in the option
+			if ( ! is_numeric( $timestamp ) ) {
+				continue;
+			}
+
+			// Skip events whose time hasn't come
+			if ( $timestamp > $current_window ) {
+				break;
+			}
+
+			// Extract just the essentials needed to retrieve the full job later on
+			foreach ( $timestamp_events as $action => $action_instances ) {
+				foreach ( $action_instances as $instance => $instance_args ) {
+					// There are some jobs we never care to run
+					if ( $this->is_blocked_event( $action ) ) {
+						wp_unschedule_event( $timestamp, $action, $instance_args['args'] );
+						continue;
+					}
+
+					// Queue internal events separately to avoid them being blocked
+					$queue = $this->is_internal_event( $action ) ? 'internal_events' : 'current_events';
+
+					array_push( $$queue, array(
+						'timestamp' => $timestamp,
+						'action'    => md5( $action ),
+						'instance'  => $instance,
+					) );
+				}
+			}
+		}
+
+		// Limit batch size to avoid resource exhaustion
+		if ( count( $current_events ) > $this->job_queue_size ) {
+			$current_events = array_slice( $current_events, 0, $this->job_queue_size );
+		}
+
+		return rest_ensure_response( array(
+			'events'          => $current_events,
+			'internal_events' => $internal_events,
+			'endpoint'        => get_rest_url( null, $this->namespace . '/event/' ),
+		) );
+	}
+
+	/**
+	 * Execute a specific event
+	 */
+	public function run_event( $request ) {
+		// Parse request
+		$event     = $request->get_json_params();
+		$timestamp = isset( $event['timestamp'] ) ? absint( $event['timestamp'] ) : null;
+		$action    = isset( $event['action'] ) ? trim( sanitize_text_field( $event['action'] ) ) : null;
+		$instance  = isset( $event['instance'] ) ? trim( sanitize_text_field( $event['instance'] ) ) : null;
```

Comment:
`$index` will work. To differentiate jobs with the same timestamp and action, WP passes the event's arguments through `md5` and uses that to index the cron array, which is how I arrived at `$instance`.